packages/react/src/useTable.tsx

```diff
@@ -248,10 +149,105 @@ const getRows = (props: { data: GadgetRecord<any>[]; columns: TableColumn[] }) =
   const { columns, data } = props;
 
   return data.map((record) => {
-    const row: Record<string, ColumnValueType> = { id: record.id };
+    const row: TableRow = { id: record.id };
     for (const { apiIdentifier, getValue } of columns) {
       row[apiIdentifier] = getValue(record);
     }
     return row;
   });
 };
+
+const getTableColumnArray = (fields: FieldMetadataFragment[], columnsFromOptions: TableOptions["columns"]) => {
+  const columnsMap =
```

Comment:
I know this was pre-existing, but I think this name is a bit unclear, maybe it could be 'relatedColumnFieldMap or something like that?